cmd/nginx-ingress/main.go

```diff
@@ -58,43 +58,90 @@ func main() {
 
 	parseFlags()
 
-	config, kubeClient := createConfigAndKubeClient()
+	config, err := getClientConfig()
+	if err != nil {
+		glog.Fatalf("error creating client configuration: %v", err)
+	}
 
-	kubernetesVersionInfo(kubeClient)
+	kubeClient, err := getKubeClient(config)
+	if err != nil {
+		glog.Fatalf("Failed to create client: %v.", err)
+	}
 
-	validateIngressClass(kubeClient)
+	err = confirmMinimumK8sVersionCriteria(kubeClient)
+	if err != nil {
+		glog.Fatal(err)
+	}
```

Comment:
Since the error that is used here is never re-used outside of the scope of the return, you can also format the check like this. This is more of a `go-lang` way of doing things. It's useful when you want reduce the scope of the return value (in this case the error)
```suggestion
	if err := confirmMinimumK8sVersionCriteria(kubeClient); err != nil {
		glog.Fatal(err)
	}
```